### Loading the Website Traffic Data

In [2]:
import pandas as pd

# Loading the dataset
wt = pd.read_csv("website_wata.csv")

# Display basic information about the dataset
wt.head()


Page Views  Session Duration  Bounce Rate Traffic Source  Time on Page  \
0           5         11.051381     0.230652        Organic      3.890460   
1           4          3.429316     0.391001         Social      8.478174   
2           4          1.621052     0.397986        Organic      9.636170   
3           5          3.629279     0.180458        Organic      2.071925   
4           5          4.235843     0.291541           Paid      1.960654   

   Previous Visits  Conversion Rate  
0                3              1.0  
1                0              1.0  
2                2              1.0  
3                3              1.0  
4                5              1.0

In [5]:
wt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Page Views        2000 non-null   int64  
 1   Session Duration  2000 non-null   float64
 2   Bounce Rate       2000 non-null   float64
 3   Traffic Source    2000 non-null   object 
 4   Time on Page      2000 non-null   float64
 5   Previous Visits   2000 non-null   int64  
 6   Conversion Rate   2000 non-null   float64
dtypes: float64(4), int64(2), object(1)
memory usage: 109.5+ KB


### Overview of Our DataFrame

#### From the above info, we can deduce that the dataset contains a total of 2,000 records and 7 columns, which include 1 categorical and 7 numerical variables.

Numerical Variables:

* `Page Views` (int)
* `Session Duration` (float)
* `Bounce Rate` (float)
* `Time on Page` (float)
* `Previous Visits` (int)
* `Conversion Rate` (float)

Categorical Variable:

* `Traffic Source` (object/string)

### Summary Statistics

In [4]:
wt.describe()

Page Views  Session Duration  Bounce Rate  Time on Page  \
count  2000.000000       2000.000000  2000.000000   2000.000000   
mean      4.950500          3.022045     0.284767      4.027439   
std       2.183903          3.104518     0.159781      2.887422   
min       0.000000          0.003613     0.007868      0.068515   
25%       3.000000          0.815828     0.161986      1.935037   
50%       5.000000          1.993983     0.266375      3.315316   
75%       6.000000          4.197569     0.388551      5.414627   
max      14.000000         20.290516     0.844939     24.796182   

       Previous Visits  Conversion Rate  
count      2000.000000      2000.000000  
mean          1.978500         0.982065  
std           1.432852         0.065680  
min           0.000000         0.343665  
25%           1.000000         1.000000  
50%           2.000000         1.000000  
75%           3.000000         1.000000  
max           9.000000         1.000000

### Traffic Sources Value Count

In [13]:
# Unique values in categorical column (Traffic Source) and the number of times they appear
traffic_sources = wt["Traffic Source"].value_counts().sort_values(ascending=False)

traffic_sources

Traffic Source
Organic     786
Paid        428
Referral    301
Social      269
Direct      216
Name: count, dtype: int64

### Key Observations from the Summary Statistics and Categorical Data:

##### User Engagement Insights:

* The average session duration is ~3.02 minutes, but it varies significantly (with a standard deviation of ~3.10).
* The average time on page is ~4.03 minutes, but the max value (of ~24.8 minutes) suggests outliers.

##### Bounce Rate and Conversion Rate:

* The average bounce rate is ~28.5%, with some sessions having a bounce rate as low as 0.8% and as high as 84.5%.
* The conversion rate is high (98.2% on average), indicating strong lead generation or goal completions.

##### `Traffic Sources` Breakdown:

* The majority of traffic comes from Organic Search (786 visits), followed by Paid (428 visits) and Referral (301 visits).
Direct traffic (216 visits) is the least common source, while Social (269) is the least but one.